numpy/_core/tests/test_multiarray.py

```diff
@@ -9803,18 +9803,94 @@ def test_null_inside_ustring_array_is_truthy(self):
 class TestFormat:
 
     def test_0d(self):
+        # for now we are converting to Python type and calling its format
+        # ref: numpy/core/src/multiarray/scalartypes.c.src:gentype_format
         a = np.array(np.pi)
         assert_equal(f'{a:0.3g}', '3.14')
         assert_equal(f'{a[()]:0.3g}', '3.14')
 
+    def test_general_format(self):
+        a = np.array([np.pi])
+        with pytest.raises(NotImplementedError):
+            f'{a:.4g}'
+
+        # FIXME: When the format character is implemented, the following
+        #        tests should succeed.
+        return
```

Comment:
Maybe this, and then indent the tests on 9820-9835?

```suggestion
        try:
            f'{a:.4g}'
        except NotImplementedError:
            # FIXME: When the format character is implemented, the following
            pass
        else:
```